### PR TITLE
fix: refactor prompts model

### PR DESCRIPTION
**Pull request by kappa90**
## Problem

Prompts are not showing to any users except on the PostHog team, because `request.user.distinct_id` is only set for the Cloud team, not for other teams, and we're using Person in the SequenceState model.

## Changes

- refactored the model to use User rather than Person
- refactored the api endpoint

## How did you test this code?
- Made sure tests are still running

### CODE REVIEW

**Review by benjackwhite — APPROVED** [review]
I think this makes sense for now given that is a test but clearly shows how different the actual implementation will have to be

**Comment by kappa90** [comment]
> I think this makes sense for now given that is a test but clearly shows how different the actual implementation will have to be

Yes, the thing is that we should have an endpoint like /decide to retrieve the prompts, not an endpoint behind the authentication flow as it is now.

File: posthog/api/prompt.py [hunk 1]
```diff
@@ -34,20 +34,22 @@ def my_prompts(self, request: request.Request, **kwargs):
             )
             local_states.append(parsed_state)
 
+        team_id = 2  # we are testing prompts created by us, on PostHog Cloud
```
Comment (Twixes) [inline posthog/api/prompt.py:37]: This constant kinda looks like a magic number, could be a bit clearer by naming it `POSTHOG_CLOUD_PRODUCTION_TEAM_ID` or something like that

File: posthog/api/prompt.py [hunk 2]
```diff
@@ -34,20 +34,22 @@ def my_prompts(self, request: request.Request, **kwargs):
...
+
         my_prompts: Dict[str, Any] = {"sequences": [], "state": {}}
         states_to_update: List[PromptSequenceState] = []
         states_to_create: List[PromptSequenceState] = []
 
         person_id = (
-            PersonDistinctId.objects.filter(distinct_id=request.user.distinct_id, team_id=self.team_id)
+            PersonDistinctId.objects.filter(distinct_id=request.user.distinct_id, team_id=team_id)
```
Comment (Twixes) [inline posthog/api/prompt.py:44]: Note that usage of the hard-coded team ID makes this unusable on self-hosted instances. Do we have guards ensuring that this branch of the code can only run on Cloud?

**Review by neilkakkar — COMMENTED** [review]
Would love for us to make a decision here: https://posthog.slack.com/archives/C0113360FFV/p1659615400513919?thread_ts=1659608497.319549&cid=C0113360FFV - before moving forward here :)

TL;DR: We should link to the `User` model, not the `Person` model, which unlocks this for self hosted too.

**Comment by kappa90** [comment]
@Twixes @neilkakkar I refactored the model and the api to use User rather than Person, can you please check if this implementation makes sense? Ty!

File: posthog/migrations/0255_user_prompt_sequence_state.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+                ("id", models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name="ID")),
+                ("key", models.CharField(max_length=400)),
+                ("last_updated_at", models.DateTimeField(default=django.utils.timezone.now)),
+                ("step", models.IntegerField(default=0)),
+                ("completed", models.BooleanField(default=False)),
+                ("dismissed", models.BooleanField(default=False)),
+                ("user", models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to=settings.AUTH_USER_MODEL)),
+            ],
+        ),
+        migrations.DeleteModel(name="PromptSequenceState",),
```
Comment (neilkakkar) [inline posthog/migrations/0255_user_prompt_sequence_state.py:28]: Code looks alright to me, although this specific migration can give you problems. Before this goes in, ensure that:

1. The code before-this-PR can run when this migration has run (to prevent downtime on cloud) (Imagine a situation on cloud where the migration has run, but the code hasn't been updated yet. This happens because multiple web instances.)
2. Running this migration doesn't lock other tables which brings down other parts of the app querying those tables.

Comment (kappa90) [reply posthog/migrations/0255_user_prompt_sequence_state.py:28]: Thanks, I'll consult with other people to be sure the migration doesn't do any of that 🙏

Comment (neilkakkar) [reply posthog/migrations/0255_user_prompt_sequence_state.py:28]: s/works/doesn't crash PostHog - it's _fine_ if it returns nothing for the 10-15 minutes it takes to completely get on the new version, since this is new / not mission critical / doesn't have much data yet.

**Comment by kappa90** [comment]
@pauldambra ready for a final check

File: posthog/models/prompt.py [hunk 4]
```diff
@@ -4,14 +4,11 @@
 from django.utils import timezone
 
 
-class PromptSequenceState(models.Model):
```
Comment (pauldambra) [inline posthog/models/prompt.py:7]: This rename leaves a pending model deletion migration. I think we need to add `UserPromptSequence` not rename `PromptSequence`

**Review by pauldambra — CHANGES_REQUESTED** [review]

File: posthog/models/prompt.py (hunk 4 above)
Comment (kappa90) [reply posthog/models/prompt.py:7]: I pushed a fix while you were typing this :D

**Review by pauldambra — APPROVED** [review]
